vendor-extra/JatsContentBundle/tests/EventListener/BuildView/FrontContribAuthorAffContentHeaderListenerTest.php

```diff
@@ -0,0 +1,232 @@
+            </contrib>
+        </contrib-group>
+    </article-meta>
+</front>
+XML
+        );
+
+        $event = new BuildViewEvent($element, new TemplateView('template'));
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
+        $this->assertSame('template', $view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_there_is_no_aff() : void
+    {
+        $listener = new FrontContribAuthorAffContentHeaderListener($this->createFailingConverter());
+
+        $element = $this->loadElement(
+            <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<front xmlns="http://jats.nlm.nih.gov">
+    <article-meta>
+        <contrib-group>
+            <contrib contrib-type="author"/>
+        </contrib-group>
+    </article-meta>
+</front>
+XML
+        );
+
+        $event = new BuildViewEvent($element, new TemplateView('@LiberoPatterns/content-header.html.twig'));
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
+        $this->assertSame('@LiberoPatterns/content-header.html.twig', $view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_there_is_already_affiliations_set() : void
+    {
+        $listener = new FrontContribAuthorAffContentHeaderListener($this->createFailingConverter());
+
+        $element = $this->loadElement(
+            <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<front xmlns="http://jats.nlm.nih.gov">
+    <article-meta>
+        <contrib-group>
+            <contrib contrib-type="author">
+                <aff>foo</aff>
+            </contrib>
+        </contrib-group>
+    </article-meta>
+</front>
+XML
+        );
+
+        $event = new BuildViewEvent(
+            $element,
+            new TemplateView('@LiberoPatterns/content-header.html.twig', ['affiliations' => 'bar'])
+        );
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
+        $this->assertSame('@LiberoPatterns/content-header.html.twig', $view->getTemplate());
+        $this->assertSame(['affiliations' => 'bar'], $view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    /**
+     * @test
+     */
+    public function it_sets_the_affiliations_argument() : void
+    {
+        $listener = new FrontContribAuthorAffContentHeaderListener(
+            new CallbackViewConverter(
+                static function (NonDocumentTypeChildNode $node, ?string $template = null, array $context = []) : View {
+                    if (!$node instanceof Element
```

Comment:
extracting a single type of element as `text` seems like a common use case, is this `CallbackViewConverter` something that should be created by a superclass, to make it available to many tests?

Reply:
(similar to `createDumpingConverter`)

Reply:
Complexity here is due to requiring `text` (which is incompatible with `createDumpingConverter()` ) https://github.com/libero/browser/blob/ac23393ee2459a529e28534daea90a96597c587c/vendor-extra/JatsContentBundle/src/EventListener/BuildView/FrontContribAuthorAffContentHeaderListener.php#L48-L49

Though maybe it should fail rather than skip in that case?

Reply:
Turns out the name fails if nothing sets the `text`, so should just match.

Could do with more useful error messages though.

Reply:
Which also relates to https://github.com/libero/jats-support/pull/13#discussion_r294242059.

Reply:
Just realised

| Complexity here is due to requiring `text`

isn't true, it's due to the `array_unique`...

Reply:
Should probably move this elsewhere.

Reply:
Wrong commit name, but done in 568a3a398807eeabe08a0b84fd161e643c45618c.